### PR TITLE
Add a source JSONField to Interaction

### PR DESCRIPTION
**Pull request by currycoder**
### Description of change

Adds a field `Interaction.source' - to represent data about external sources that we may create interactions from.  This is a JSONField to allow developers to be unconstrained in how they use this field for different data sources.

### Context

This will allow us to record/look up meeting ids for interactions that are automatically created for meeting invite emails.  This will assist us in making sure that we don't create duplicate interactions for the same meeting.

### Points of note

I haven't proposed that we set a default for this field - as this might be a little complex with json fields.  It might be preferable to perform a manual migration across the DB to set every interaction to have `source={}` - this could be coupled with an API change to ensure that newly created interactions have a default source of `{}`.  Any thoughts?

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

**Comment by reupen** [comment]
> I haven't proposed that we set a default for this field - as this might be a little complex with json fields. It might be preferable to perform a manual migration across the DB to set every interaction to have source={} - this could be coupled with an API change to ensure that newly created interactions have a default source of {}. Any thoughts?

It it would help with querying, but it seems pretty forgiving to me as both of these work (don't return an error):

```
select null::jsonb @> '{"b": 2}'::jsonb;
select null::jsonb -> 'b';
```

Also, even if the field were non-nullable it wouldn't preclude strings and numbers as values.

So, null as the empty value seems OK to me (and nothing would stop us changing our minds later on). Are there any plans for the front end to use `source`?

File: changelog/interaction/source.db.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+A supporting field was added to ``interaction_interaction`` for the 
+purpose of allowing interactions to be archived:
```
Comment (reupen) [inline changelog/interaction/source.db.rst]: Is this the right text for this news fragment?

**Comment by currycoder** [comment]
> At the moment the field is editable in the admin site, should we make it read-only and format the same way we did for the user event log?

Good idea - I think this makes a lot of sense.  I can't see a use-case for admins to adjust the source on an interaction through the admin panel.

File: changelog/interaction/source.api.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+``GET /v3/interaction`` and ``GET /v3/interaction/<uid>``: The following field was added:
+
+* ``source`` - object, nullable - object representing the external source of
+  an interaction, e.g. a calendar invite
+
+This field is read only.
```
Comment (marcofucci) [inline changelog/interaction/source.api.rst]: Why does the frontend need this field?

Comment (currycoder) [reply changelog/interaction/source.api.rst]: Good point.  It doesn't for any of the work that is currently in scope, I suppose I didn't know what our approach should be in this case.  Should what we expose be on a "need to know" basis in general?

File: changelog/interaction/source.db.rst (hunk 1 above)
Comment (currycoder) [reply changelog/interaction/source.db.rst]: Good spot.  copypasta.

**Comment by currycoder** [comment]
> So, null as the empty value seems OK to me (and nothing would stop us changing our minds later on). Are there any plans for the front end to use `source`?

Thanks for thinking about this.  I'll leave it as null for now.  There are no plans for frontend to use source, and as per the comment from @marcofucci I might just remove it from the API entirely.

**Pull request merged**